k6/k8s.js

```diff
@@ -3,86 +3,102 @@ import { sleep } from 'k6';
+export function create(url, body, params = null) {
+  const res = http.post(url, JSON.stringify(body), params);
 
-    check(res, {
-        'POST returns status 201 or 409': (r) => r.status === 201 || r.status === 409,
-    })
+  check(res, {
+    'POST returns status 201 or 409': (r) => r.status === 201 || r.status === 409,
+  })
 
-    if (res.status === 409) {
-        // wait a bit and try again
-        sleep(Math.random())
+  if (res.status === 409) {
+    // wait a bit and try again
+    sleep(Math.random())
 
-        return create(url, body)
-    }
+    return create(url, body, params)
+  }
 
-    return res
+  return res
 }
 
 // deletes a k8s resource
-export function del(url){
-    const res = http.del(url)
+export function del(url) {
+  const res = http.del(url)
 
-    check(res, {
-        'DELETE returns status 200 or 404': (r) => r.status === 200 || r.status === 404,
-    })
+  check(res, {
+    'DELETE returns status 200 or 404': (r) => r.status === 200 || r.status === 404,
+  })
 
-    return res
+  return res
 }
 
 const continueRegex = /"continue":"([A-Za-z0-9]+)"/;
 
 // lists k8s resources
-export function list(url) {
-    let _continue = 'first'
-    let responses = []
+export function list(url, params = null) {
+  let _continue = 'first'
+  let responses = []
+
+  while (_continue != null) {
+    const fullUrl = new URL(url);
+    fullUrl.searchParams.append('limit', limit);
+    fullUrl.searchParams.append('timeout', timeout);
+    fullUrl.searchParams.append('watch', false);
+    if (_continue !== 'first') {
+      fullUrl.searchParams.append('continue', _continue);
+    }
 
-    while (_continue != null) {
-        const fullUrl = new URL(url);
-        fullUrl.searchParams.append('limit', limit);
-        fullUrl.searchParams.append('timeout', timeout);
-        fullUrl.searchParams.append('watch', false);
-        if (_continue !== 'first') {
-            fullUrl.searchParams.append('continue', _continue);
-        }
+    const res = http.get(fullUrl.toString(), params);
 
-        const res = http.get(fullUrl.toString());
+    check(res, {
+      'list returns status 200': (r) => r.status === 200,
+    });
 
-        check(res, {
-            'list returns status 200': (r) => r.status === 200,
-        });
+    const found = res.body.match(continueRegex);
+    _continue = found !== null ? found[1] : null
 
-        const found = res.body.match(continueRegex);
-        _continue = found !== null ? found[1] : null
+    responses.push(res)
+  }
 
-        responses.push(res)
-    }
+  return responses
+}
+
+// patches k8s resources and valid subresources
+export function patch(url, body, params = null) {
+  // patch requires these headers to be set to appropriate values, below are the defaults
+  let defaultParams = { headers: { "Content-Type": "application/merge-patch+json", "Accept": "application/json" } }
+  if (!params.hasOwnProperty("headers")) _.merge(params, defaultParams)
```

Comment:
IIUC the only reason to pull in lodash (and the whole nodejs dependency tree) seems to be this line.

I would encourage using the standard library and get rid of the lodash import and package.json/package-lock.json:

```suggestion
if (!params.hasOwnProperty("headers")) {
    params = Object.assign(params, defaultParams)
}
```